src/components/CurrentLocation.jsx

```diff
@@ -0,0 +1,120 @@
+import { useState } from 'react';
+import { useEffect } from 'react';
+import React, { useContext } from 'react';
+import axios from 'axios';
+import { AuthenticationContext, TemperatureContext } from '../Context';
+import Loader from '../components/Loader';
+
+class CurrentLocation extends React.Component {
+  static authenticated = AuthenticationContext;
+  static temperature = TemperatureContext;
+
+  constructor() {
+    super();
+    this.state = {
+      userLocation: '',
+      userCoords: ''
+    };
+  }
+
+  updateIcon(data) {
+    document.getElementById('icon-weather').src =
+      'https://developer.foreca.com/static/images/symbols/' + this.state.userLocation.symbol + '.png';
+  };
+
+  componentDidMount() {
+    const that = this;
+
+    if ('geolocation' in navigator) {
+      navigator.geolocation.getCurrentPosition(function (position) {
+        const userCoordinates = {
+          latLon: position.coords.longitude + ',' + position.coords.latitude
+        };
+
+        (async function getPin() {
+          await axios
+            .post('http://localhost:3000/get-current-location-params', userCoordinates, {
+              headers: { 'Content-Type': 'application/json; charset=UTF-8' }
+            })
+            .then(data => that.setState({ userCoords: data.data }));
+        })();
+      });
+    } else {
+      alert('Location Not Available');
+    }
+  }
+
+  componentDidUpdate(prevProps, prevState) {
+    if (prevState.userCoords == '') {
```

Comment:
in case you need to use equality, please use strict one according to best practices
```suggestion
    if (!prevState.userCoords) {
```